openpype/hosts/blender/plugins/publish/extract_fbx_animation.py

```diff
@@ -56,7 +71,10 @@ def process(self, instance):
             starting_frames.append(curr_frame_range[0])
             ending_frames.append(curr_frame_range[1])
         else:
-            self.log.info("Object have no animation.")
+            self.log.info(
+                f"Armature '{armature.name}' has no animation, "
+                f"skipping FBX animation extraction for {instance}."
+            )
             return
 
         asset_group_name = asset_group.name
```

Comment:
We need some more changes here! Some of them are unrelated to the PR, but it's a very quick fix that is necessary to test it so I think we can include them here. (I'll write them all here because GitHub doesn't allow me to put comments in line that don't have changes)

This is necessary because this property is here only if you load from a json layout, not from blend (it's my overlook, and I don't know how we missed before 😅)
```
        asset_name = asset_group.get(AVALON_PROPERTY).get("asset_name")
        if asset_name:
            asset_group_name = asset_group.name
            asset_group.name = asset_name
```

We have then to remove `asset_group.select_set(True)` and `asset_group.select_set(False)`, because it was an object before, but now it is a collection and it cannot be selected. We also need to remove `asset_group` in the override context (we should also change the override to use the new way to override the context, but maybe that's for another PR).
```
        # asset_group.select_set(True)
        armature.select_set(True)
        fbx_filename = f"{instance.name}_{armature.name}.fbx"
        filepath = os.path.join(stagingdir, fbx_filename)

        override = plugin.create_blender_context(
            active=armature, selected=[armature])
        bpy.ops.export_scene.fbx(
            override,
            filepath=filepath,
            use_active_collection=False,
            use_selection=True,
            bake_anim_use_nla_strips=False,
            bake_anim_use_all_actions=False,
            add_leaf_bones=False,
            armature_nodetype='ROOT',
            object_types={'EMPTY', 'ARMATURE'}
        )
        armature.name = armature_name
        if asset_name:
            asset_group.name = asset_group_name
        # asset_group.select_set(False)
        armature.select_set(False)

```

I think this is the minimum to make it work, but in the future it needs some serious rework!

Reply:
Does this work without selecting a "root" node? Is selecting the "armature" sufficient? I feel like we might be lacking export data then?

Reply:
Does this work https://github.com/ynput/OpenPype/pull/5910/commits/480c37533b8ce0e542e7b052ae28ff4d7a775acb @simonebarbieri ?

Reply:
It does! That's great :D Thanks!